model/histogram/generic.go

```diff
@@ -393,7 +406,44 @@ func checkHistogramBuckets[BC BucketCount, IBC InternalBucketCount](buckets []IB
 	return nil
 }
 
-func getBound(idx, schema int32) float64 {
+func checkHistogramCustomBounds(bounds []float64, spans []Span) error {
+	prev := math.Inf(-1)
+	for _, curr := range bounds {
+		if curr <= prev {
+			return fmt.Errorf("previous bound is %f and current is %f: %w", prev, curr, ErrHistogramCustomBucketsInvalid)
+		}
+		prev = curr
+	}
+
+	var totalSpanLength int
+	for _, span := range spans {
```

Comment:
Add a check that the span.Offset cannot be negative. (The function `checkHistogramSpans` does not check this condition for the first span, but here we're indexing an array so we need the check.)